exporters/stats/signalfx/src/test/java/io/opencensus/exporter/stats/signalfx/SignalFxStatsExporterTest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2017, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.exporter.stats.signalfx;
+
+import io.opencensus.common.Duration;
+import java.util.Properties;
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for {@link SignalFxStatsExporter}. */
+@RunWith(JUnit4.class)
+public class SignalFxStatsExporterTest {
+
+  private static final String TEST_TOKEN = "token";
+  private static final String TEST_ENDPOINT = "https://example.com";
+  private static final Duration ONE_SECOND = Duration.create(1, 0);
+
+  @Rule public final ExpectedException thrown = ExpectedException.none();
+
+  @After
+  public void tearDown() {
+    System.getProperties().remove(SignalFxStatsExporter.SIGNALFX_HOST_PROPERTY);
+    System.getProperties().remove(SignalFxStatsExporter.SIGNALFX_TOKEN_PROPERTY);
+    SignalFxStatsExporter.unsafeResetExporter();
```

Comment:
I think unsafeResetExporter() needs to be called after each test cases that creates an exporter.

Reply:
Not sure I see your point. That's exactly what this `@After` method does. For tests where the exporter isn't created (expected failure), calling `unsafeResetExporter()` will simply be a no-op.

Reply:
Ahh that's right, I mixed up with `@After`, `@AfterEach` and `@AfterClass`.